### PR TITLE
Close tally scopes on service stop

### PR DESCRIPTION
**Pull request by glibsm**
Fix the root metrics scope to use the correct interface also (which is the one that has `Close()` method) and call it on shutdown.

GFM-96 #resolve

### CODE REVIEW

**Comment by glibsm** [comment]
@uber-go/service-framework

File: core/metrics/metrics_test.go [hunk 1]
```diff
@@ -66,15 +66,15 @@ func TestRegisterBadReporterPanics(t *testing.T) {
...
 	return tally.NoopScope, nil
 }
 
-func badScope(i ScopeInit) (tally.Scope, error) {
+func badScope(i ScopeInit) (tally.RootScope, error) {
 	return nil, errors.New("fake error")
 }
 
-func getScope() tally.Scope {
+func getScope() tally.RootScope {
```
Comment (sectioneight) [inline core/metrics/metrics_test.go]: Does NopScope conform to RootScope? What's the difference in the interfaces?

File: service/host.go [hunk 2]
```diff
@@ -134,16 +134,18 @@ func (s *host) shutdown(err error, reason string, exitCode *int) (bool, error) {
...
+	if len(errs) > 0 {
+		for k, v := range errs {
+			ulog.Logger().Error("Failure to shut down module", "name", k.Name(), "error", v.Error())
+		}
+	}
 
-	// }
+	// Stop the metrics reporting
+	if s.scope != nil {
+		s.scope.Close()
```
Comment (sectioneight) [inline service/host.go]: Does this flush immediately, or do we need to wait on a channel or something?

Comment (glibsm) [inline service/host.go]: `Close()` is the only difference

File: core/metrics/metrics_test.go (hunk 1 above)
Comment (glibsm) [inline core/metrics/metrics_test.go]: NoopScope is a RootScope

Comment (sectioneight) [inline core/metrics/metrics_test.go]: Sweet.

**Comment by sectioneight** [comment]
Why the drop in coverage?

**Comment by glibsm** [comment]
Probably due to un-commented code that does failure logging, will double check. Not sure we can test that yet

**Pull request merged**